### PR TITLE
[Shop] Possibility to disable the e-mail verification after registration

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: features/account/registering.feature [hunk 1]
```diff
@@ -50,3 +63,16 @@ Feature: Account registration
         When I register with email "ghastly@bespoke.com" and password "suitsarelife"
         Then I should be notified that new account has been successfully created
         And a welcoming email should have been sent to "ghastly@bespoke.com"
+
+    @ui @email
+    Scenario: Receiving a verification email after registration
+        When I register with email "ghastly@bespoke.com" and password "suitsarelife"
+        Then I should be notified that new account has been successfully created
+        And a verification email should have been sent to "ghastly@bespoke.com"
```
Comment (michalmarcinkowski) [inline features/account/registering.feature]: This scenario is already covered (see [here](https://github.com/michalmarcinkowski/Sylius/blob/0ce1d75d208bdbd265199d04f5c7939318d94f21/features/account/verification/verifying_email_address.feature#L49-L49)).

File: src/Sylius/Behat/Context/Ui/EmailContext.php [hunk 2]
```diff
@@ -70,6 +70,22 @@ public function aWelcomingEmailShouldHaveBeenSentTo($recipient)
...
+        $this->assertEmailContainsMessageTo('To verify your email address', $recipient);
+    }
+
+    /**
+     * @Then a verification email should not have been sent to :recipient
+     */
+    public function aVerificationEmailShouldNotHaveBeenSentTo($recipient)
+    {
+        $this->assertEmailNotContainsMessageTo('To verify your email address', $recipient);
+    }
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Ui/EmailContext.php]: These steps should be unified with the scenarios [here](https://github.com/michalmarcinkowski/Sylius/blob/0ce1d75d208bdbd265199d04f5c7939318d94f21/features/account/verification/verifying_email_address.feature).

File: src/Sylius/Bundle/ShopBundle/EventListener/UserRegistrationListener.php [hunk 3]
```diff
@@ -82,4 +117,18 @@ private function handleUserVerificationToken(ShopUserInterface $user)
 
         $this->eventDispatcher->dispatch(UserEvents::REQUEST_VERIFICATION_TOKEN, new GenericEvent($user));
     }
+
+    /**
+     * @param ShopUserInterface $user
+     */
+    private function verifyAndLogin(ShopUserInterface $user)
+    {
+        $user->setVerifiedAt(new \DateTime());
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ShopBundle/EventListener/UserRegistrationListener.php]: IMO this feature should only enable the user account and allow for a login, but not automatically verify the email address. The verification email should be sent anyway, so the user can confirm that the email address belongs to him.

Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/EventListener/UserRegistrationListener.php]: I understood this task slightly differently, that this feature enables account, user is logged after registration and there is no need to send verification email... But of course, I will change that :)

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu!

File: src/Sylius/Bundle/CustomerBundle/Resources/translations/flashes.en.yml [hunk 4]
```diff
@@ -3,4 +3,4 @@
 
 sylius:
     customer:
-        register: 'Thank you for registering, check your email to verify your account.'
+        register: 'Thank you for registering. If you are not logged in, check your email to verify your account.'
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CustomerBundle/Resources/translations/flashes.en.yml]: This should be reverted. You always receive verification email.

**Pull request merged**